src/main/java/uk/gov/hmcts/darts/courthouse/service/CourthouseServiceImpl.java

```diff
@@ -115,7 +121,48 @@ public List<ExtendedCourthouse> mapFromEntitiesToExtendedCourthouses(List<Courth
 
     @Override
     public List<CourthouseEntity> getAllCourthouses() {
-        return courthouseRepository.findAll();
+        List<Integer> userCourthouseIds = new ArrayList<>();
+
+        UserAccountEntity userAccount = authorisationApi.getCurrentUser();
```

Comment:
shouls this not just use the `authorisationApi.getListOfCourthouseIdsUserHasAccessTo()` method?
then make a call to `courthouseRepository` with those courthouseIds?

Reply:
Although looking at the current finder `UserRolesCourthousesRepository#findAllCourthouseIdsByUserAccount` 
It feels a bit strange to me - I think we could have a return type of List<CourthouseEntity> with query having a `SELECT DISTINCT urc.courthouse`
Since if you look at the `UserRolesCourthousesEntity#courthouse` it will be currently eagerly fetching the `CourthouseEntity` anyway?

Reply:
when `UserRolesCourthousesRepository#findAllCourthouseIdsByUserAccount` was first made, i made it for the AdvancedSearch, and i only needed the courthouseIds as it was being used in the search, so there was no need to bring back the full courthouse object as that would mean further calls to the database. 
AdvancedSearch has now been enhanced to use a different view, so this method is no longer used by it, so i guess can be modified as required.

Reply:
Code and tests updated

Reply:
none of the code from lines 126 to 142 is needed.

Reply:
In the case where Admin or Judge is provided this coded is required.

Reply:
code updated as per suggestion